grammar.js

```diff
@@ -326,49 +326,49 @@ module.exports = grammar({
 
     _function_value_parameter: $ => seq(
       optional($.parameter_modifiers),
-      $.parameter,
-      optional(seq("=", $._expression))
+      field('parameter', $.parameter),
+      optional(seq("=", field('initializer', $._expression)))
     ),
 
     _receiver_type: $ => seq(
       optional($.type_modifiers),
-      choice (
+      field('type', choice (
         $._type_reference,
         $.parenthesized_type,
         $.nullable_type
-      )
+      ))
     ),
 
     function_declaration: $ => prec.right(seq( // TODO
       optional($.modifiers),
       "fun",
       optional($.type_parameters),
-      optional(seq($._receiver_type, optional('.'))),
-      $.simple_identifier,
-      $._function_value_parameters,
-      optional(seq(":", $._type)),
+      optional(seq(field('receiver', $._receiver_type), optional('.'))),
+      field('name', $.simple_identifier),
+      field('parameters', $._function_value_parameters),
+      optional(seq(":", field('return_type', $._type))),
       optional($.type_constraints),
-      optional($.function_body)
+      optional(field('body', $.function_body))
     )),
 
-    function_body: $ => choice($._block, seq("=", $._expression)),
+    function_body: $ => choice(field('block', $._block), seq("=", field('expression', $._expression))),
 
     variable_declaration: $ => prec.left(PREC.VAR_DECL, seq(
       // repeat($.annotation), TODO
-      field("name", $.simple_identifier),
-      optional(seq(":", field ("type", $._type)))
+      field('name', $.simple_identifier),
+      optional(seq(":", field('type', $._type)))
     )),
 
     property_declaration: $ => prec.right(seq(
       optional($.modifiers),
       choice("val", "var"),
       optional($.type_parameters),
-      optional(seq($._receiver_type, optional('.'))),
-      choice($.variable_declaration, $.multi_variable_declaration),
+      optional(seq(field('receiver_type', $._receiver_type), optional('.'))),
```

Comment:
But here (in `optional('.')`), the author used single quotes, so maybe you just picked one.